src/openloco/windows/stationwnd.cpp

```diff
@@ -1,55 +1,660 @@
+            const auto y = self->y + widget.top - 1;
+            const auto width = widget.width();
+            gfx::draw_string_494BBF(*dpi, x, y, width, colour::black, string_ids::buffer_1250, &args);
+        }
+
+        // 0x0048EB0B
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::caption:
+                    common::renameStationPrompt(self, widgetIndex);
+                    break;
+
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_station:
+                case common::widx::tab_cargo:
+                case common::widx::tab_cargo_ratings:
+                    common::switchTab(self, widgetIndex);
+                    break;
+
+                case widx::station_catchment:
+                    if (self->number == word_112C786)
+                        self->number = 65535;
+
+                    // 0x0049271A
+                    registers regs;
+                    regs.ax = self->number;
+                    call(0x0049271A, regs);
+                    break;
+            }
+        }
+
+        // 0x0048EBB7
+        static void on_resize(window* self)
+        {
+            // Call to sub_48E32C has been deliberately omitted.
+
+            self->set_size(gfx::ui_size_t(299, 172), gfx::ui_size_t(299, 327));
+        }
+
+        // 0x0048EB64
+        static void get_scroll_size(window* self, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+        {
+            auto station = stationmgr::get(self->number);
+            *scrollHeight = 0;
+            for (uint8_t i = 0; i < 32; i++)
+            {
+                if (station->cargo_stats[i].quantity != 0)
+                {
+                    *scrollHeight += 12;
+                    if (station->cargo_stats[i].origin != self->number)
+                        *scrollHeight += 10;
+                }
+            }
+        }
+
+        // 0x0048EB4F
+        static void tooltip(FormatArguments& args, ui::window* window, widget_index widgetIndex)
+        {
+            args.push(string_ids::tooltip_scroll_cargo_list);
+        }
+
+        // 0x0048E986
+        static void draw_scroll(window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+        {
+            gfx::clear_single(*dpi, colour::get_shade(self->colours[1], 4));
+
+            const auto station = stationmgr::get(self->number);
+            int16_t y = 1;
+            for (int i = 0; i < 32; i++)
+            {
+                auto& cargo = station->cargo_stats[i];
+                auto quantity = cargo.quantity;
+                if (quantity == 0)
+                    continue;
+
+                if (quantity > 400)
+                    quantity = 400;
+
+                quantity += 9;
+                quantity /= 10;
```

Comment:
You could create a new variable called units which comprise of how many tens of quantity there are.